posts/2021/06/pytorch-ignite-distributed.rst

```diff
@@ -0,0 +1,460 @@
+
+Thus, you will now be able to run the same version of the code across
+all supported backends seamlessly:
+
+-  backends from native torch distributed configuration:
+   `nccl <https://github.com/NVIDIA/nccl>`__,
+   `gloo <https://github.com/facebookincubator/gloo>`__,
+   `mpi <https://www.open-mpi.org/>`__
+
+-  `Horovod <https://horovod.readthedocs.io/en/stable/>`__ framework
+   with ``gloo`` or ``nccl`` communication backend
+
+-  XLA on TPUs via `pytorch/xla <https://github.com/pytorch/xla>`__
+
+In this blog post we will compare PyTorch-Ignite's API with torch
+native's distributed code and highlight the differences and ease of use
+of the former. We will also show how Ignite's ``auto_*`` methods
+automatically make your code compatible with the aforementioned
+distributed backends so that you only have to bring your own model,
+optimizer and data loader objects.
+
+Code snippets, as well as commands for running all the scripts, are
+provided in a separate
+`repository <https://github.com/pytorch-ignite/idist-snippets>`__.
+
+Then we will also cover several ways of spawning processes via torch
+native ``torch.multiprocessing.spawn`` and also via multiple distributed
+launchers in order to highlight how Pytorch-Ignite's ``idist`` can
+handle it without any changes to the code, in particular:
+
+-  `torch.multiprocessing.spawn <https://pytorch.org/docs/stable/multiprocessing.html#torch.multiprocessing.spawn>`__
+
+-  `torch.distributed.launch <https://pytorch.org/docs/stable/distributed.html#launch-utility>`__
+
+-  `horovodrun <https://horovod.readthedocs.io/en/stable/running_include.html>`__
+
+-  `slurm <https://slurm.schedmd.com/>`__
+
+More information on launchers experiments can be found
+`here <https://github.com/sdesrozis/why-ignite>`__.
+
+.. _-pytorch-ignite-unified-distributed-api:
+
+🔥 Pytorch-Ignite Unified Distributed API 
+===========================================
+
+We need to write different code for different distributed backends. This
+can be tedious especially if you would like to run your code on
+different hardware configurations. Pytorch-Ignite's ``idist`` will do
+all the work for you, owing to the high-level helper methods.
+
+.. _-focus-on-the-helper-auto-methods:
+
+🔍 Focus on the helper ``auto_*`` methods:
+---------------------------------------------
+
+-  `auto_model() <https://pytorch.org/ignite/distributed.html#ignite.distributed.auto.auto_model>`__
+
+This method adapts the logic for non-distributed and available
+distributed configurations. Here are the equivalent code snippets for
+distributed model instantiation:
+
+.. raw:: html
+
+   <embed>
+      <div>
+         <table>
+            <tr>
+               <th style="text-align:center;">PyTorch-Ignite</th>
+               <th style="text-align:center;">PyTorch DDP</th>
+            </tr>
+            <tr>
+               <td colspan="2"> <img src="/images/pytorch-ignite/distributed-made-easy-with-ignite/ignite_vs_ddp_automodel.png"> </td>
```

Comment:
As it is rendered in github, image  is not aligned with the columns above. This seems to be the case for the for the rest of the images related to the auto_ section.

Reply:
@fco-dv, please use https://deploy-preview-208--quansight-labs.netlify.app/blog/2021/06/distributed-made-easy-with-ignite/ to see the final rendering

Reply:
ok thanks